### PR TITLE
VIDEO-10736 WebGL Pipeline for Cross Browser Support

### PR DESCRIPTION
**Pull request by charliesantos**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Pull Request Details

- Check changelog for details of the changes
- All links will be updated after RC, then again before release
- `twilio-video.js` file will be removed and replaced with the CDN link once video sdk is released
- Feel free to ignore `lib/processors/webgl2/` folder. But check `lib/processors/webgl2/index.ts` for the note
- Future work to consider if time permits
  - Rewrite `webgl2` module so it conforms with our coding standards
  - Separate out `webgl2` and `canvas2d` pipelines
  - Update example app UI to play with config values

### Before review
* [x] Updated CHANGELOG.md if necessary
* [x] Added unit tests if necessary
* [x] Updated affected documentation
* [x] Verified locally with `npm test`
* [x] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [x] Ready for review

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,29 @@
 1.1.0 (In Progress)
```
Comment (manjeshbhargav) [inline CHANGELOG.md:1]: This should be `2.0.0`

File: lib/processors/webgl2/helpers/webglHelper.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+ * to get GLSL syntax highlighting.
+ * https://marketplace.visualstudio.com/items?itemName=boyswan.glsl-literal
+ *
+ * On VSCode OSS, boyswan.glsl-literal requires slevesque.shader extension
+ * to be installed as well.
+ * https://marketplace.visualstudio.com/items?itemName=slevesque.shader
+ */
+export const glsl = String.raw
+
+export function createPiplelineStageProgram(
```
Comment (manjeshbhargav) [inline lib/processors/webgl2/helpers/webglHelper.ts:12]: Can we make sure the webgl files follow our programming conventions, like semicolons at the end of the line, etc.?

File: lib/utils/support.ts [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
+}
+
 /**
  * @private
  */
 export function isBrowserSupported() {
-  if (typeof window !== 'undefined') {
-    return !!window.OffscreenCanvas && !(/Mobi/i.test(window.navigator.userAgent)) && !!window.chrome;
+  if (typeof window !== 'undefined' && typeof document !== 'undefined') {
+    return (!!getCanvas().getContext('2d') || !!getCanvas().getContext('webgl2'));
```
Comment (manjeshbhargav) [inline lib/utils/support.ts:13]:
```suggestion
    return !!(getCanvas().getContext('2d') || getCanvas().getContext('webgl2'));
```

**Review by manjeshbhargav — APPROVED** [review]

File: lib/processors/webgl2/helpers/webglHelper.ts (hunk 2 above)
Comment (charliesantos) [reply lib/processors/webgl2/helpers/webglHelper.ts:12]: It's one of my notes in the PR which I lowered in priority due to time constraints.

**Pull request merged**